backend/routes/UserRoutes.js

```diff
@@ -142,4 +147,48 @@ router.put("/edit/:id", async (req, res, next) => {
   }
 });
 
+router.patch("/set-roles/:id", validate(["roles"], []), async (req, res) => {
+  const roles = json5.parse(req.body.roles);
+  const keyroles = [
+    "Wish Granter",
+    "Volunteer",
+    "Mentor",
+    "Airport Greeter",
+    "Office",
+    "Special Events",
+    "Translator",
+    "Speaker's Bureau",
+    "Las Estrellas",
+    "Primary Admin",
+    "Secondary Admin",
+  ];
+  if (JSON.stringify(Object.keys(roles)) === JSON.stringify(keyroles)) {
+    const roles_user = [];
+    Object.keys(roles).forEach(async (key) => {
+      if (roles[key]) {
+        await roles_user.push(key);
+      }
+    });
+    await UserModel.findByIdAndUpdate(req.params.id, { roles: roles_user }).catch((error) => {
+      console.log(error);
+      res.status(400).json({ error: "User not found" });
+    });
+  } else {
+    res.status(400).json({ erorr: "Malformed request" });
+  }
```

Comment:
I think a better way to do this would be something like (with the newer route format we've started using):

```js
UserModel.findById(req.user._id)
  .then((user) => {
    roles.forEach((role) => {
      if(user.roles.indexOf(role) === -1) user.roles.push(role);
    });
    return user.save();
  })
  .then(() => res.json({ success: true }))
  .catch(errorHandler(res));
```

This way lets the user add roles one at a time, rather than all at once.

Reply:
Done, and understood! Will take care of this newer format for next time :)